teps/0033-proxy-support.md

```diff
@@ -0,0 +1,306 @@
+---
+title: proxy-support
+
+authors:
+  - "@piyush-garg"
+  - "@vdemeester"
+  
+creation-date: 2020-11-02
+
+last-updated: 2020-11-02
+
+status: proposed
+
+---
+
+# TEP-0033: Proxy Support
+
+<!-- toc -->
+- [Summary](#summary)
+- [Motivation](#motivation)
+  - [Goals](#goals)
+  - [Non-Goals](#non-goals)
+- [Requirements](#requirements)
+- [Proposal](#proposal)
+  - [User Stories (optional)](#user-stories-optional)
+  - [Notes/Constraints/Caveats (optional)](#notesconstraintscaveats-optional)
+  - [Risks and Mitigations](#risks-and-mitigations)
+  - [User Experience (optional)](#user-experience-optional)
+  - [Performance (optional)](#performance-optional)
+- [Design Details](#design-details)
+  - [Proposed Solutions](#proposed-solutions)
+- [Test Plan](#test-plan)
+- [Drawbacks](#drawbacks)
+- [Alternatives](#alternatives)
+- [Infrastructure Needed (optional)](#infrastructure-needed-optional)
+- [Upgrade &amp; Migration Strategy (optional)](#upgrade--migration-strategy-optional)
+- [References (optional)](#references-optional)
+<!-- /toc -->
+
+## Summary
+
+The proposal is to provide the proxy support to tektoncd/pipelines and tektoncd/triggers.
+
+Tekton should provide a way to use network proxy settings defined for the platform so that workloads
+can make use of it during execution.
+
+The proposal is designed considering the way proxy settings at the container level is used. It just uses the
+container env spec which can be specified using taskrun/pipelinerun and leverage that. It abstracts the way to
+specify env for pods/containers to the user.
+
+This proposal covers the proxy environment settings for following components:
+
+1. Controller
+2. Workloads like taskruns and pipelineruns
+
+## Motivation
+
+Support users or organisations that require a network proxy settings for their workload to communicate to outside world, including CI/CD.
+
+### Goals
+
+1. Workloads executions like taskrun/pipelinerun should use global proxy settings.
+2. Controllers should use global proxy settings for communicating.
+
+### Non-Goals
+
+1. Managing or maintaining proxy settings and servers.
+
+## Requirements
+
+1. User should be able to define proxy config for all workloads
+2. User should be able to provide proxy config for specific taskrun/pipelinerun
+3. Controllers should use proxy setting for communication
+
+## Proposal
+
+This proposal is to allow tektoncd/pipeline and tektoncd/triggers to have proxy support
+enabled / configurable to benefit users/organizations to use proxy feature to communicate to
+the outside world of the platform.
+
+### User Stories (optional)
+
+User should be able to define global proxy settings on the platform and tektoncd/pipeline
+and tektoncd/triggers act based on the proxy settings defined.
+
+1. User A has http proxy setup on cluster.
+2. User B has http/https proxy setup on cluster along with certs.
```

Comment:
So I see mention of `cert` here, but then I'm missing it elsewhere (at least my serach of `cert` just got one hit.

Still reading ....

Reply:
But I don't see a "we won't support certs initially" disclaimer in the non goals